File: recovery_assessment.m
function v = recovery_assessment(Z, Z_hat)
%RECOVERY_ASSESSMENT
%   v = recovery_assessment(Z, Z_hat)
%

assert(length(Z) == size(Z_hat, 3))

% Shorthand.
R = length(Z);

v = 0;
for i = 1:R
  v = v + 1/R*norm(Z_hat(:, :, i) - Z{i}, 'fro');
end

end

File: recovery_assessment.py
from math import sqrt
from numpy.linalg import norm

def recovery_assessment(Z, Z_hat):
  assert(len(Z) == Z_hat.shape[2])

  num = 0
  den = 0

  for p in range(len(Z)):
    num += norm(Z_hat[:,:,p] - Z[p])**2
    den += norm(Z[p])**2

  return sqrt(num) / sqrt(den)

if __name__ == '__main__':
  from loadmat import loadmat
  mat = loadmat('20180124/singlegraph_bss_logdet_knownx_02bJ6ZgjQVAdrHUsO6xJ.mat')
  print(recovery_assessment(mat['truth']['Z'], mat['Z_hat']))
